llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -201,35 +204,31 @@ def clean_output(
         return output
 
     def postprocess_output(self, cfg, df: pd.DataFrame, output: Dict) -> Dict:
-        output = self.clean_output(output, self.prompts, cfg)
+        if not cfg.prediction.metric == "Perplexity":
+            output = self.clean_output(output, self.prompts, cfg)
 
         output["target_text"] = self.answers
-        metric_func, _ = cfg.prediction.metric_class.get(cfg.prediction.metric)
-        predictions = output["predicted_text"]
-        labels = output["target_text"]
-        assert len(predictions) == len(labels)
+        if "predicted_text" not in output.keys():
+            # In perplexity metric, no predictions were made, use the target text
+            output["predicted_text"] = output["target_text"]
```

Comment:
I'd not set predicted text at all. I can confuse user who download predictions while being unaware that perplexity metric triggers no text generation.